detection_rules/rule_validators.py

```diff
@@ -40,12 +40,20 @@ def validate(self, data: QueryRuleData, meta: RuleMeta) -> None:
             packages_manifest = load_integrations_manifests()
             package_integrations = TOMLRuleContents.get_packaged_integrations(data, meta, packages_manifest)
 
+            validation_checks = {"stack": None, "integrations": None}
             # validate the query against fields within beats
-            self.validate_stack_combos(data, meta)
+            validation_checks["stack"] = self.validate_stack_combos(data, meta)
 
             if package_integrations:
                 # validate the query against related integration fields
-                self.validate_integration(data, meta, package_integrations)
+                validation_checks["integrations"] = self.validate_integration(data, meta, package_integrations)
+
+            if("Error" in str(type(validation_checks["stack"])) and not package_integrations):
```

Comment:
Why do we need to cast to `type`? What is the type of `validation_checks["stack"]`? It might be better to compare against the actual error type to be more explicit with is `instance()`